app/src/main/java/de/westnordost/streetcomplete/osm/surface/Surface.kt

```diff
@@ -0,0 +1,46 @@
+package de.westnordost.streetcomplete.osm.surface
+
+import de.westnordost.streetcomplete.R
+
+enum class Surface(val osmValue: String) {
+    ASPHALT("asphalt"),
+    CONCRETE("concrete"),
+    CONCRETE_PLATES("concrete:plates"),
+    CONCRETE_LANES("concrete:lanes"),
+    FINE_GRAVEL("fine_gravel"),
+    PAVING_STONES("paving_stones"),
+    COMPACTED("compacted"),
+    DIRT("dirt"),
+    SETT("sett"),
+    UNHEWN_COBBLESTONE("unhewn_cobblestone"),
+    GRASS_PAVER("grass_paver"),
+    WOOD("wood"),
+    WOODCHIPS("woodchips"),
+    METAL("metal"),
+    GRAVEL("gravel"),
+    PEBBLES("pebblestone"),
+    GRASS("grass"),
+    SAND("sand"),
+    ROCK("rock"),
+    CLAY("clay"),
+    ARTIFICIAL_TURF("artificial_turf"),
+    TARTAN("tartan"),
+    PAVED_ROAD("paved"),
+    UNPAVED_ROAD("unpaved"),
+    GROUND_ROAD("ground"),
+    PAVED_AREA("paved"),
+    UNPAVED_AREA("unpaved"),
+    GROUND_AREA("ground"),
+
+    // extra values, recording duplicates
+    MUD("mud"), // valid value, but not displayed specially in StreetComplete
+    EARTH("earth"), // synonym of "dirt", maybe more clear
+    // this values ideally would be removed from OpenStreetMap, but while they remain
+    // we want to handle them somehow
+    SOIL("soil"),
+    PAVING_STONES_WITH_WEIRD_SUFFIX("paving_stones:30"),
+    COBLLESTONE_FLATTENED("cobblestone:flattened"),
+    BRICK("brick"),
+    BRICKS("bricks");
```

Comment:
that puts annoyance in other places: it is no longer possible to check `old_surface != new_surface` for example in check for pending changes, debugging gets more confusing and there are some exotic undo scenarios where data can be lost in unexpected ways.

reasoning about app state started to be confusing so I decided that a tiny bit duplication is much nicer than checking for synonimity in already confusing places.

Reply:
> it is no longer possible to check old_surface != new_surface

Why not? If `element.tags["surface"]` was `bricks`, then in the overlay, you'd have 
```
val originalSurface: Surface = PAVING_STONES
```

Reply:
I tried to implement this and it feels fiddly in ugly way - though for different reasons than I remembered.

Instead of slight duplication in some `when` constructs and compiler-enforced handling of this value it is necessary to convert them first one way and then remember to convert them back.

Both solutions are a bit fiddly and I really prefer one where compiler will hold my hand - and it is not even so cumbersome.

Potential code is in https://github.com/streetcomplete/StreetComplete/commit/04bd2d6f8629fa279dd93e9ba0555e2deb006465 (not tested yet, and I am not fan of testing that thoroughly enough to be convinced that I have not missed anything)